src/targets.rs

```diff
@@ -697,6 +702,36 @@ impl FromStr for OperatingSystem {
     type Err = ();
 
     fn from_str(s: &str) -> Result<Self, ()> {
+        if s.starts_with("macosx") {
+            // Parse operating system names like `macosx10.7.0`.
+            let s = &s["macosx".len()..];
+            let mut parts = s.split(".").map(|num| num.parse::<u16>());
+
+            macro_rules! get_part {
+                () => {
+                    if let Some(Ok(part)) = parts.next() {
+                        part
+                    } else {
+                        return Err(());
+                    }
+                };
+            }
+
+            let major = get_part!();
+            let minor = get_part!();
+            let patch = get_part!();
+
+            if parts.next().is_some() {
+                return Err(());
+            }
+
+            return Ok(OperatingSystem::MacOSX {
+                major,
+                minor,
+                patch,
+            })
+        }
```

Comment:
I think this code is fine for now, but please add a TODO mentioning that we should generalize this to support other OS's, such as `Ios` and `Darwin`, which also have version numbers.

Reply:
Done